test/zhr_devs/bakery_integration/commands/gen_multiplayer_test.exs

```diff
@@ -0,0 +1,125 @@
+defmodule ZhrDevs.BakeryIntegration.Commands.GenMultiplayerTest do
+  use ExUnit.Case, async: true
+
+  @moduletag :capture_log
+
+  alias ZhrDevs.BakeryIntegration.Commands.GenMultiplayer
+
+  alias Uptight.Text, as: T
+
+  use Witchcraft.Comonad
+
+  import ExUnit.CaptureLog
+
+  describe "output_json_path/1" do
+    test "with atom task - returns expected path" do
+      assert GenMultiplayer.output_json_path(:task) =~ "tournament/task/output.json"
+    end
+
+    test "with string task - returns expected path" do
+      assert GenMultiplayer.output_json_path("task") =~ "tournament/task/output.json"
+    end
+  end
+
+  describe "build/1" do
+    test "with valid opts - returns expected result" do
+      opts = [
+        submissions_folder: T.new!("submissions_folder"),
+        server_code: T.new!("server_code"),
+        task: :on_the_map_goo
+      ]
+
+      assert [
+               {:cmd, %Ubuntu.Command{}},
+               {:output_json_path, output_json_path}
+             ] = opts |> GenMultiplayer.build() |> Uptight.Result.from_ok()
+
+      assert output_json_path =~ "tournament/on_the_map_goo/output.json"
```

Comment:
I'm not a huge fan that we have a bit of a mess between just using raw string (binary), something called Path and Ubuntu.Path, which is kinda the best.